### PR TITLE
Auto-generate zero data for missing data payload

### PR DESCRIPTION
**Pull request by bjlittle**
This pull-request extends the capability of the `iris_grib.message._DataProxy` to detect and auto-generate missing data payload for a GRIB2 message.

The scenario is that the message data payload is all zero, and rather than serialize this within the message, thus increasing the footprint, an application can generate the data at runtime whenever the Data Representation Section [5] has `bitsPerValue=0` and an empty Data Section [7].

Failure to support this extension results in no encoded Data Section [7] within a `iris_grib.message._RawGribMessage` causing a `KeyError` traceback within the user script. The impact of this is that an N-D Iris cube can be loaded but whenever at least one GRIB2 message has such missing data, touching the cube data will result in this failure.

Reference: https://github.com/SciTools/iris-test-data/pull/89

### CODE REVIEW

**Comment by bjlittle** [comment]
@pp-mo As a result of this fix, we now have "unexpected successes" in the tests regarding pickling.

We could now simply not mark these tests are expected failures. 

How would you like to proceed?

**Comment by bjlittle** [comment]
Out of curiosity, I remove the `@unittest.expectedFailure` decorators ...

File: iris_grib/message.py [hunk 1]
```diff
@@ -237,14 +237,29 @@ def _bitmap(self, bitmap_section):
...
     def __getitem__(self, keys):
-        # NB. Currently assumes that the validity of this interpretation
+        # N.B. Assumes that the validity of this interpretation
         # is checked before this proxy is created.
 
         message = self.recreate_raw()
         sections = message.sections
+        data = None
+
+        if 5 in sections:
```
Comment (bjlittle) [inline iris_grib/message.py:247]: This defensive check is necessary, as we have tests with no encoded section 5 entries in the message.

Otherwise we generate a `KeyError` when attempting an unprotected `sections[5]` look-up.

**Comment by pp-mo** [comment]
> We could now simply not mark these tests are expected failures.

After some chat, I think this is OK.
I'll do some back-fill to check what the error USED to be, but I suspect we just found that latest-eccodes has fixed it (= made the pickling possible)

File: iris_grib/tests/integration/load_convert/test_data_section.py [hunk 2]
```diff
@@ -68,5 +69,16 @@ def test_grid_complex_spatial_differencing(self):
         self.assertCMLApproxData(cube)
 
 
+class TestDataProxy(tests.IrisGribTest):
+    def test_data_representation__no_bitsPerValue(self):
+        """Ensures that zero data is auto-generated."""
+        path = tests.get_data_path(
```
Comment (pp-mo) [inline iris_grib/tests/integration/load_convert/test_data_section.py:75]: Can we just add a comment, that this test file contains an empty data section ?

**Review by pp-mo — APPROVED** [review]
Unless you want to follow a tiny suggestion.. 
good to go I think

File: iris_grib/tests/integration/load_convert/test_data_section.py (hunk 2 above)
Comment (bjlittle) [reply iris_grib/tests/integration/load_convert/test_data_section.py:79]: @pp-mo See https://github.com/SciTools/iris-grib/pull/362/commits/92435d865722e54382c7d530b189bc6a46d0b703

Comment (pp-mo) [reply iris_grib/tests/integration/load_convert/test_data_section.py:79]: Wow, that's comprehensive !

**Comment by bjlittle** [comment]
@pp-mo Sweet, cheers 💯
